core/src/main/scala/cats/derived/contravariant.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2015 Miles Sabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats
+package derived
+
+import shapeless._
+import util.VersionSpecific.OrElse
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("Could not derive an instance of Contravariant[${F}]")
+trait MkContravariant[F[_]] extends Contravariant[F] {
+  def safeContramap[A, B](fa: F[A])(f: B => Eval[A]): Eval[F[B]]
+  def contramap[A, B](fa: F[A])(f: B => A): F[B] = safeContramap(fa)(f andThen Eval.now).value
+}
+object MkContravariant extends MkContravariantDerivation {
+  def apply[F[_]](implicit F: MkContravariant[F]): MkContravariant[F] = F
+}
+private[derived] abstract class MkContravariantDerivation extends MkContravariantNested {
+  implicit val mkContraHNil: MkContravariant[Const[HNil]#λ] = mkContraConst
+  implicit val mkContraCNil: MkContravariant[Const[CNil]#λ] = mkContraConst
+
+  implicit def mkContraConst[T]: MkContravariant[Const[T]#λ] =
+    new MkContravariant[Const[T]#λ] {
+      def safeContramap[A, B](t: T)(f: B => Eval[A]): Eval[T] = Eval.now(t)
+    }
+}
+
+private[derived] abstract class MkContravariantNested extends MkContravariantCons {
+  implicit def mkContravariantFunctorNested[F[_]](implicit F: Split1[F, Functor, ContraOrMk]): MkContravariant[F] =
```

Comment:
```suggestion
  implicit def mkFunctorContravariantNested[F[_]](implicit F: Split1[F, Functor, ContraOrMk]): MkContravariant[F] =
```
I think we are still missing (hence the suggestion):
```scala
implicit def mkContravariantFunctorNested[F[_]](implicit F: Split1[F, ContraOrMk, Functor]): MkContravariant[F]
```